### PR TITLE
Add loading state to Board Page

### LINKED ISSUE #253: Add loading state to Board Page
## **User story** 
**Who:** As a developer
**What:** I want to create a loading state to check the candidates state.
**Why:** So that BoardPage will be able to refresh dynamically when a candidate added or deleted.

### PR DESCRIPTION
**Pull request by Rasim-Deniz**
# Description

Added loading state to Board Page to control the added or deleted candidates.

Fixes #253 

# How to test?

npm run dev

# Checklist

- [X] I have performed a self-review of my own code
- [X] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [X] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [X] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [X] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Comment by Rasim-Deniz** [comment]
@senner007 Niels, I tried to implement following the way you mentioned but I'm getting this error at the moment. Could you please have a look what is missing? 
adafaadas

**Comment by Rasim-Deniz** [comment]
> @senner007 Niels, I tried to implement following the way you mentioned but I'm getting this error at the moment. Could you please have a look what is missing?
> adafaadas

This error is gone but I don't get the candidates now.

**Comment by Saidemm** [comment]
> @senner007 Niels, I tried to implement following the way you mentioned but I'm getting this error at the moment. Could you please have a look what is missing?
> adafaadas

The error is fixed. It seems it doesn't need to send the boardId as a prop in resultButton.

**Comment by Rasim-Deniz** [comment]
> > @senner007 Niels, I tried to implement following the way you mentioned but I'm getting this error at the moment. Could you please have a look what is missing?
> > adafaadas
> 
> This error is gone but I don't get the candidates now.

Everything is fixed thanks to Hanieh and Saide! @senner007 Please take a look and let us merge 😄

**Review by senner007 — CHANGES_REQUESTED** [review]
This duplicates the logic in the other board pr

**Comment by Rasim-Deniz** [comment]
> This duplicates the logic in the other board pr

Conflicts are solved. Can you check if there is any problem? We need this.

File: src/client/containers/BoardPage/BoardPageComponents/CandidateList/CandidateListPostDeadline.js [hunk 1]
```diff
@@ -4,12 +4,8 @@ import { CardItemDecorator } from '../../../../components/CandidateCard/Candidat
 import { SortableItem } from '../../../DragAndSortAdapter/DragAndSortAdapter';
 import PropTypes from 'prop-types';
 
-export default function CandidateListPostDeadline({
-  userId,
-  boardId,
-  displayDelete,
-}) {
-  const { candidates, error } = useCandidates(userId, boardId);
+export default function CandidateListPostDeadline({ displayDelete }) {
```
Comment (senner007) [inline src/client/containers/BoardPage/BoardPageComponents/CandidateList/CandidateListPostDeadline.js:7]: You are still showing the error here. In the other pr you removed it after Slavias review

File: src/client/containers/UseHooks/useCandidates.js [hunk 2]
```diff
@@ -19,13 +26,14 @@ export function useCandidates(userId, boardId) {
             };
           }),
         );
+        setIsCandidateLoading(false);
       } else {
         setError(
```
Comment (senner007) [inline src/client/containers/UseHooks/useCandidates.js]: Same thing with the ApiError here also

**Review by senner007 — CHANGES_REQUESTED** [review]
few changes

**Review by senner007 — APPROVED** [review]
✔

**Pull request merged**